### PR TITLE
[cmake] explicitly set ENABLE_EXPORTS

### PR DESCRIPTION
**Pull request by stefansaraev**
## Description

I was playing around with [CMP0065](https://cmake.org/cmake/help/v3.4/policy/CMP0065.html) set to NEW globally in my cross toolchain, and figured out that kodi.bin needs -rdynamic for our dll wrapper to work
```
/usr/lib/kodi/kodi.bin: symbol lookup error: /usr/lib/kodi/system/libcpluff-aarch64.so: undefined symbol: dllmalloc
```

## Motivation and Context

make sure ENABLE_EXPORTS is set. if we ever switch to cmake_minimum_required(VERSION 3.4) (or earlier, in case cmake people remove the OLD behaviour of the policy), we need it anyway

## How Has This Been Tested?

build and runtime tested on linux with -DCMAKE_POLICY_DEFAULT_CMP0065=NEW

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -496,6 +496,8 @@ foreach(msg ${final_message})
   message(STATUS ${msg})
 endforeach()
 
+set_target_properties(${APP_NAME_LC} PROPERTIES ENABLE_EXPORTS ON)
```
Comment (wsnipex) [inline CMakeLists.txt]: please move up to L357

Comment (stefansaraev) [inline CMakeLists.txt]: did you mean for everything but android or before the CORE_SYSTEM_NAME check?

**Review by wsnipex — APPROVED** [review]
thanks!

**Comment by wsnipex** [comment]
jenkins build and merge

**Pull request merged**